### PR TITLE
[a11y] Table header for articles category list

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #.

### Summary of Changes
To be accessible a table must have a header. 
See for examle this checklist: https://dequeuniversity.com/checklists/web/tables

Joomla provides a param show_headings for frontend tables. This removes the table header. 

This PR sets the table header visually-hidden so it is still present for assistive technologies. 

I open some similar PRs to make testing easier. 

### Testing Instructions
You need at least one article 
Make a menu Item of type category list.
Set param table_headings in tab "list layouts" to "hide"
See the view and sourcecode ot the table in frontend 

### Actual result BEFORE applying this Pull Request
There is no table heading visible, also in source code.

### Expected result AFTER applying this Pull Request
There is no table heading visible, but it is in sourcecode with class="visually-hidden"

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_content/tmpl/category/default_articles.php [hunk 1]
```diff
@@ -123,49 +123,48 @@
             <caption class="visually-hidden">
                 <?php echo Text::_('COM_CONTENT_ARTICLES_TABLE_CAPTION'); ?>
             </caption>
-            <?php if ($this->params->get('show_headings')) : ?>
-                <thead>
-                    <tr>
-                        <th scope="col" id="categorylist_header_title">
-                            <?php echo HTMLHelper::_('grid.sort', 'JGLOBAL_TITLE', 'a.title', $listDirn, $listOrder, null, 'asc', '', 'adminForm'); ?>
+            <thead <?php echo $this->params->get('show_headings', '1') ? '' :  'class="visually-hidden"'; ?>>
```
Comment (brianteeman) [inline components/com_content/tmpl/category/default_articles.php]: Not sure its that important but this code will result in `<thead >`
But the suggested code will produce `<thead>`
```suggestion
            <thead<?php echo $this->params->get('show_headings', '1') ? '' :  ' class="visually-hidden"'; ?>>
```

**Comment by ChristineWk** [comment]
I have tested this item :white_check_mark: successfully on 09286951af135cadfb973685a9480b04e741c350<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40137">issues.joomla.org/tracker/joomla-cms/40137</a>.</sub>

**Comment by ChristineWk** [comment]
`<caption class="visually-hidden">
                Articles            </caption>
            <thead class="visually-hidden">`

File: components/com_content/tmpl/category/default_articles.php [hunk 2]
```diff
@@ -123,49 +123,48 @@
             <caption class="visually-hidden">
                 <?php echo Text::_('COM_CONTENT_ARTICLES_TABLE_CAPTION'); ?>
             </caption>
-            <?php if ($this->params->get('show_headings')) : ?>
-                <thead>
-                    <tr>
-                        <th scope="col" id="categorylist_header_title">
-                            <?php echo HTMLHelper::_('grid.sort', 'JGLOBAL_TITLE', 'a.title', $listDirn, $listOrder, null, 'asc', '', 'adminForm'); ?>
+            <thead<?php echo $this->params->get('show_headings', '1') ? '' :  ' class="visually-hidden"'; ?>>
```
Comment (Quy) [inline components/com_content/tmpl/category/default_articles.php]:
```suggestion
            <thead<?php echo $this->params->get('show_headings', '1') ? '' : ' class="visually-hidden"'; ?>>
```

File: components/com_content/tmpl/category/default_articles.php [hunk 3]
```diff
@@ -123,49 +123,48 @@
...
+                        <th scope="col" id="categorylist_header_ratings">
+                            <?php echo HTMLHelper::_('grid.sort', 'COM_CONTENT_RATINGS', 'rating', $listDirn, $listOrder); ?>
+                        </th>
+                    <?php endif; ?>
+                    <?php if ($isEditable) : ?>
+                        <th scope="col" id="categorylist_header_edit"><?php echo Text::_('COM_CONTENT_EDIT_ITEM'); ?></th>
+                    <?php endif; ?>
+                </tr>
+            </thead>
+
```
Comment (Quy) [inline components/com_content/tmpl/category/default_articles.php]:
```suggestion
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ae99a8512644259faff97cbbefc1f26cedfe677d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40137">issues.joomla.org/tracker/joomla-cms/40137</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40137">issues.joomla.org/tracker/joomla-cms/40137</a>.</sub>

**Comment by obuisard** [comment]
Thank you Christiane @chmst for the PR. Better accessibility, yes!

**Pull request merged**